src/capture/capture.cpp

```diff
@@ -1422,28 +1422,23 @@ void Capture::NotifyMetaIfNextVideo::notify_next_video(
 
 namespace
 {
-    std::string get_session_id_basename(const char *basename) noexcept
-    {        
+    std::string get_session_id_basename(const char *basename)
+    {
         std::size_t session_id_size = 0;
         
         while (basename[session_id_size] && basename[session_id_size] != ',')
         {
             ++session_id_size;
         }
-
-        std::string session_id;
-
-        session_id.reserve(session_id_size);
-        session_id.assign(basename, session_id_size);
-        return session_id;
+        return std::string(basename, session_id_size);
     }
 }
 
 Capture::Capture(
     const CaptureParams& capture_params,
     const DrawableParams& drawable_params,
     bool capture_wrm, const WrmParams& wrm_params,
-    bool capture_png, const PngParams& png_params,
+    bool capture_png, PngParams& png_params,
```

Comment:
Why did `const` disappear ?

Reply:
Because you told me "real_basename should be in PngParam", so I need to remove const for assign "session_id.c_str()" in "real_basename" variable of png_params instance ^^

Reply:
Then `png_params.real_basename` should be initialized from outside, `png_params.rt_basename_only_sid` should not exist and `get_session_id_basename(...)` replaced by `ini.get<cfg::context::session_id>()`.